packages/core/src/AlertBar/AlertBar.js

```diff
@@ -20,110 +20,133 @@ import { Message } from './Message.js'
         }
     }
-
-    hide = event => {
-        clearTimeout(this.displayTimeout)
-        this.setState({ visible: false })
-
-        this.onHiddenTimeout = setTimeout(() => {
-            this.setState(
-                { hidden: true },
-                () => this.props.onHidden && this.props.onHidden({}, event)
+    const resumeDisplayTimeout = () => {
+        if (shouldAutoHide) {
+            clearAllTimeouts()
+            displayTimeout.current = setTimeout(
+                runHideAnimation,
+                displayTimeRemaining.current
             )
-        }, ANIMATION_TIME)
-    }
-
-    shouldAutoHide() {
-        const { permanent, warning, critical } = this.props
-        return !(permanent || warning || critical)
+        }
     }
 
-    render() {
-        const {
-            className,
-            children,
-            success,
-            warning,
-            critical,
-            icon,
-            actions,
-            dataTest,
-        } = this.props
-        const { visible, hidden } = this.state
-
-        if (hidden) {
-            return null
+    useEffect(() => {
+        // Additional check on inDOM prevents the AlertBar from briefly showing
+        // when it is mounted with a hidden prop set to true
+        if (hidden && inDOM) {
+            runHideAnimation()
+        }
+        if (!hidden) {
+            runShowAnimation()
         }
 
-        const info = !critical && !success && !warning
-
-        return (
-            <div
-                className={cx(className, {
-                    info,
-                    success,
-                    warning,
-                    critical,
-                    visible,
-                })}
-                data-test={dataTest}
-                onMouseEnter={this.stopDisplayTimeOut}
-                onMouseLeave={this.startDisplayTimeout}
-            >
-                <Icon
-                    dataTest={`${dataTest}-icon`}
-                    icon={icon}
-                    critical={critical}
-                    success={success}
-                    warning={warning}
-                    info={info}
-                />
-                <Message>{children}</Message>
-                <Actions
-                    actions={actions}
-                    hide={this.hide}
-                    dataTest={dataTest}
-                />
-                <Dismiss onClick={this.hide} dataTest={`${dataTest}-dismiss`} />
-
-                <style jsx>{styles}</style>
-            </div>
-        )
-    }
+        return clearAllTimeouts
+    }, [hidden])
+
+    return !inDOM ? null : (
+        <div
+            className={cx(className, {
+                info,
+                success,
+                warning,
+                critical,
+                inViewport,
+            })}
+            data-test={dataTest}
+            onMouseEnter={pauzeDisplayTimeout}
+            onMouseLeave={resumeDisplayTimeout}
```

Comment:
I know this is out of scope for this PR but it'd be nice if we also paused the timer when the component is focused in order to make alerts keyboard accessible.

Reply:
Great suggestion. Could you create a new JIRA issue for this, and post a link to it here, so we don't forget?

Reply:
https://jira.dhis2.org/browse/LIBS-190

Reply:
Thanks!